transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -311,19 +317,24 @@ private void invokeChannelRead(Object msg) {
         }
     }
 
+    private void invokeNextChannelRead(AbstractChannelHandlerContext next, Object msg) {
+        nextChannelReadInvoked = true;
```

Comment:
If I understood correctly, `nextChannelReadInvoked` can be updated by any non-I/O thread, because this method is being executed by the potentially different executor than the executor of the current context.